tests/callbacks/test_finetuning_callback.py

```diff
@@ -275,7 +275,7 @@ def configure_optimizers(self):
     model = FreezeModel()
     cb = OnEpochLayerFinetuning()
     trainer = Trainer(max_epochs=10, resume_from_checkpoint=chk.last_model_path, callbacks=[cb])
-    with pytest.raises(IndexError, match="index 6 is out of range"):
+    with pytest.raises(ValueError, match="loaded state dict has a different number of parameter groups"):
```

Comment:
why has this changed? How is it related to the changes in the PR?

Reply:
This is because we are now loading the training state, including the optimizer state, earlier in the run phase. 
Previously, this would fail in a IndexError in the on_train_epoch_start of the Finetune callback. Now, it is failing earlier in the run phase when loading the optimizer states. The reason is the same, the layers have changed so the optimizer states can't be loaded 1:1.